tests/testthat/testAnnotate.R

```diff
@@ -0,0 +1,51 @@
+# testAnnotate.R
+#
+#
+context("Gene graph annotation function")
+
+# Unknown: "ENSP00000005257", "ENSP00000003084"
+names = c("ENSP00000000233", "ENSP00000005340", "ENSP00000003100")
+heats = c(0.3, 0.5, 0.2)
+test_gH <- new.env(hash = TRUE)
+for (i in 1:length(names)) {
+    test_gH[[names[i]]] <- heats[i]
+}
+
+test_gG <- importNet.STRING(fName = "dataString.txt",
+                       net = "experimental",
+                       dropUnmapped = FALSE,
+                       silent = TRUE,
+                       writeLog = FALSE)
+
+test_that("annotateGraph rejects inappropriate objects", {
+    # test null gH object
+    null_gH <- NULL
+    expect_error(annotateGraph(gH = null_gH, gG = test_gG), "Object \"null_gH\" is NULL.")
+
+    # test null gG object
+    null_gG <- NULL
+    expect_error(annotateGraph(gH = test_gH, gG = null_gG), "Object \"null_gG\" is NULL.")
+
+    # test gH object is missing
+    expect_error(annotateGraph(gG = test_gG))
+
+    # test gG object is missing
+    expect_error(annotateGraph(gH = test_gH))
+})
+
+test_that("annotateGraph has added a new \"heat\" attribute", {
+    test_AGG <- annotateGraph(gH = test_gH, gG = test_gG)
+    # make sure it exists
+    expect_true(!is.null(test_AGG))
+    expect_true(!is.null(igraph::vertex_attr(test_AGG)$heat))
```

Comment:
No - still unit testing (even though they would fail on an integration problem). Just define what a proper log message from your test-data would look like and see if you get that output from the function. 

Required for every function that contributes to the workflow, to implement our best-practice.